### DISCUSSION TITLE
Conductor UI deploying on Apache tomcat

### DISCUSSION THREAD

**Discussion by shekarpcs** [Netflix/conductor#2731] [General]
I have done yarn build on condcutor ui and copied build folder to apache tomcat. 

but ui is not able to communicate to Conductor server, instead all request going to relative path /api on tomcat server.

is there anything I need to do. 

please suggest

**Comment by peterlau** [comment]
By default the UI expects the backend API to be proxied to /api on the same host as the frontend assets. You will need to configure your hosting environment accordingly. The example here shows nginx config: https://github.com/Netflix/conductor/blob/main/docker/serverAndUI/nginx/nginx.conf

I believe you can achieve the same effect using mod_proxy `ProxyPass` in Apache. Here is the documentation: 
https://httpd.apache.org/docs/2.4/mod/mod_proxy.html

**Reply by shekarpcs** [reply]
> By default the UI expects the backend API to be proxied to /api on the same host as the frontend assets. You will need to configure your hosting environment accordingly. The example here shows nginx config: https://github.com/Netflix/conductor/blob/main/docker/serverAndUI/nginx/nginx.conf
> 
> I believe you can achieve the same effect using mod_proxy `ProxyPass` in Apache. Here is the documentation: https://httpd.apache.org/docs/2.4/mod/mod_proxy.html

Thanks for the information. I really appreciate it. 

Our organization blocked Yarn. is there a way to build using NPM. ?

**Reply by peterlau** [reply]
`npm` should work although we have not tested it. The dockerfiles are provided simply for reference - feel free to substitute the package manager. If you encounter any issues with `npm` please share it here and we can investigate.
